workshop/readme.md

```diff
@@ -36,15 +36,92 @@ docker pull ipahsd/ros2-training-foxy:01
 ```
 To test the docker image, run:
 ```
+docker run -it ipahsd/ros2-training-foxy:01 /bin/bash
+---
+root@fe7b28bad402:/#
+```
+The command creates a new container and starts it. This can be confirmed with
+```
+CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS          PORTS          NAMES
+05100b6119b1   ipahsd/ros2-training-foxy:01   "/ros_entrypoint.sh …"   8 seconds ago    Up 7 seconds                   nostalgic_ptolemy
+```
+The last column `NAMES` shows the name of the container `nostalgic_ptolemy` that is randomly assigned. These names can be manually assgined with `--name` option.
+```
+docker run -it --name test ipahsd/ros2-training-foxy:01 /bin/bash
+```
+Now to confirm the name of the container
+```
+CONTAINER ID   IMAGE                          COMMAND                  CREATED          STATUS          PORTS         NAMES
+0a52451af871   ipahsd/ros2-training-foxy:01   "/ros_entrypoint.sh …"   7 seconds ago    Up 6 seconds                  test
+```
+To stop a container without removing it
+```
+docker container stop test
+```
+To execute an existing container (container which was created with `docker run`, but later stopped with `docker stop`)
+```
+docker exec -it test /bin/bash
+```
+The command line prompt of docker (e.g. `root@fe7b28bad402:/#`) can be used like a typical Ubuntu terminal, but without graphics.
+
+The `docker run` command can be used to run `ROS` commands as well. Below is an example of [talker-listener](https://docs.ros.org/en/foxy/Tutorials/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html) by creating two containers on separate terminals and communicating between them through `ROS` messages.
+```
 docker run -it --rm ipahsd/ros2-training-foxy:01 ros2 run demo_nodes_cpp talker
```

Comment:
explain --rm?

The container will be removed automactly when existing this container